src/lib.rs

```diff
@@ -879,6 +879,51 @@ impl<'a, A, D> ArrayView<'a, A, D>
     {
         iterators::new_outer_iter(self)
     }
+
+    /// Split the array along `axis` and return one view strictly before the
+    /// split and one view after the split.
+    ///
+    /// **Panics** if `axis` is out of bounds.
+    pub fn axis_split_at(self, axis: usize, index: Ix)
+        -> (Self, Self)
+    {
+        assert!(index <= self.shape()[axis]);
+        let left_ptr = self.ptr;
+        let right_ptr = if index == self.shape()[axis] {
+            self.ptr
+        }
+        else {
+            let mut indices = self.dim.clone();
+            for (ax, ind) in indices.slice_mut().iter_mut().enumerate() {
+                if ax != axis {
+                    *ind = 0;
+                }
+                else {
+                    *ind = index;
+                }
+            }
+            let offset = self.dim.stride_offset_checked(&self.strides,
```

Comment:
On a closer look, we can just skip the whole for loop, can't we? Using stride * index directly (function `stride_offset` in dimension for example).